src/compiler/crystal/tools/init.cr

```diff
@@ -64,12 +65,29 @@ module Crystal
       fetch_required_parameter(opts, args, "NAME")
     end
 
+    def self.directory_valid?(directory)
+      if File.file?(directory)
+        puts "file #{directory} already exists"
+        return false
+      end
+
+      if Dir.exists?(directory)
+        Dir.foreach(directory) do |f|
+          unless ALLOWED_DIRECTORY_FILES.includes?(f)
+            puts "directory #{directory} is not empty (only .git/ is allowed)"
```

Comment:
Similarly: `cannot initialize Crystal project: #{directory} is not empty (only .git/ folder is allowed)`